src/coreclr/jit/morph.cpp

```diff
@@ -1219,9 +1217,7 @@ void CallArgs::ArgsComplete(Compiler* comp, GenTreeCall* call)
                 }
                 if (hasStructRegArgWeCareAbout)
                 {
-                    // Returns true if a GT_QMARK node is encountered in the argx tree
-                    //
-                    if (comp->fgWalkTreePre(&argx, Compiler::fgChkQmarkCB) == Compiler::WALK_ABORT)
+                    if (comp->gtTreeContainsOper(argx, GT_QMARK))
```

Comment:
Isn't this just dead code? All `QMARK`s should live in their own statements.

Reply:
Looks like it. Pushed some more clean up.